src/librustc/hir/mod.rs

```diff
@@ -547,10 +540,8 @@ pub enum PatKind {
     /// Such pattern can be resolved to a unit struct/variant or a constant.
     Path(Path),
 
-    /// An associated const named using the qualified path `<T>::CONST` or
-    /// `<T as Trait>::CONST`. Associated consts from inherent impls can be
-    /// referred to as simply `T::CONST`, in which case they will end up as
-    /// PatKind::Path, and the resolver will have to sort that out.
+    /// A path pattern written in qualified form, i.e. `<T as Trait>::CONST` or `<T>::CONST`.
+    /// Such patterns can only refer to associated constants at the moment.
```

Comment:
Oh, it slipped from the later unpublished part of the patch.
The old comment basically says "yep, it's an associated constant" while it's only a syntactic form - qualified path, it *can* resolve to a constant, but it also can resolve to any other associated item - a method or an associated type (?), and passes working on HIR should be ready to deal with it.